src/dds.imageio/ddsinput.cpp

```diff
@@ -183,108 +185,111 @@ ComputeNormalAG(uint8_t rgba[kBlockSize * kBlockSize * 4])
-                break;
-            case Compression::DXT2:
-            case Compression::DXT3:
-                bcdec_bc2(sourceBlock, rgbai, kBlockSize * 4);
-                break;
-            case Compression::DXT4:
-            case Compression::DXT5:
-                bcdec_bc3(sourceBlock, rgbai, kBlockSize * 4);
-                break;
-            case Compression::BC4:
-                bcdec_bc4(sourceBlock, rgbai, kBlockSize);
-                break;
-            case Compression::BC5:
-                bcdec_bc5(sourceBlock, rgbai, kBlockSize * 2);
-                break;
-            case Compression::BC6HU:
-            case Compression::BC6HS:
-                bcdec_bc6h_half(sourceBlock, rgbh, kBlockSize * 3,
-                                cmp == Compression::BC6HS);
-                break;
-            case Compression::BC7:
-                bcdec_bc7(sourceBlock, rgbai, kBlockSize * 4);
-                break;
-            default: return;
-            }
-            sourceBlock += blockSize;
-
-            /* Swap R & A for RXGB format case. */
-            if (cmp == Compression::DXT5
-                && pixelFormat.fourCC == DDS_4CC_RXGB) {
-                for (int i = 0; i < 16; ++i) {
-                    uint8_t r        = rgbai[i * 4 + 0];
-                    uint8_t a        = rgbai[i * 4 + 3];
-                    rgbai[i * 4 + 0] = a;
-                    rgbai[i * 4 + 3] = r;
-                }
-            }
-            /* Convert into full normal map if needed. */
-            else if (pixelFormat.flags & DDS_PF_NORMAL) {
-                if (cmp == Compression::BC5) {
-                    ComputeNormalRG(rgbai);
-                } else if (cmp == Compression::DXT5) {
-                    ComputeNormalAG(rgbai);
-                }
-            }
 
-            // write the pixels into the destination image location
-            if (cmp == Compression::BC6HU || cmp == Compression::BC6HS) {
-                // HDR formats: half
-                const uint16_t* pix = rgbh;
-                for (int py = 0; py < kBlockSize; ++py) {
-                    int sy        = y + py;
-                    uint16_t* dst = (uint16_t*)rgba
-                                    + channelCount * size_t(width) * sy;
-                    for (int px = 0; px < kBlockSize; ++px) {
-                        int sx = x + px;
-                        if (sx < width && sy < height) {
-                            // write the rgb value
-                            for (int ch = 0; ch < channelCount; ++ch)
-                                dst[sx * channelCount + ch] = *pix++;
-                        } else {
-                            // outside the image: skip
-                            pix += channelCount;
+    const int widthInBlocks  = (width + kBlockSize - 1) / kBlockSize;
+    const int heightInBlocks = (height + kBlockSize - 1) / kBlockSize;
+    paropt opt               = paropt(0, paropt::SplitDir::Y, 8);
```

Comment:
Oops, I just noticed that this should really be

    paropt opt = paropt(threads(), paropt::SplitDir::Y, 8);

`threads()` is an ImageInput method that returns the current thread policy. There are circumstances where somebody might ask the ImageInput not to use multiple threads (like if the app is already using threads to split up a larger body of work, and reading this file is a task of just one of them), whereas the 0 you pass will always fully utilize the thread pool.

My bad, I should have caught that. I can post a follow-up PR to fix this one little thing.

Reply:
Whoops, yeah. I was not aware of the `threads()` on image input.